### PR TITLE
Replace the URL_PATTERN regex with a simpler one

### PR DESCRIPTION
**Pull request by aermakov-zalando**
The current URL pattern is prone to DoS on some inputs. Replace it with a simpler one for now.

### CODE REVIEW

**Comment by harti2006** [comment]
:+1:

File: server/checks/Specification.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const ACTIONS = ['opened', 'edited', 'reopened', 'synchronize']
 const DEFAULT_REQUIRED_LENGTH = 8
 const ISSUE_PATTERN = /^(?:[-\w]+\/[-\w]+)?#\d+$/
 // Grubber's pattern
```
Comment (gagoman) [inline server/checks/Specification.js]: Please drop this line also

**Comment by aermakov-zalando** [comment]
👍

File: server/checks/Specification.js (hunk 1 above)
Comment (aermakov-zalando) [inline server/checks/Specification.js]: Oops, forgot to do that.

File: test/server/specification.test.js [hunk 2]
```diff
@@ -151,7 +151,7 @@ describe('Specification', () => {
 
       ['#4', 'Fix #4', 'user/repo#42', 'closes user/repo#42',
         'some-org/repo#42', 'http://tracker.com', 'https://tracker.com',
-        'www.issues.example.com', 'Fix for http://some.tracker.com/issues/42'
```
Comment (gagoman) [inline test/server/specification.test.js]: Was `www.issues.example.com` failing?

Comment (aermakov-zalando) [inline test/server/specification.test.js]: Yes. The previous regex matched URLs without the protocol (or with bogus protocols like smb://), and I'm not sure if it's really necessary. I just took the simplest possible one, which might be more lax but on the other hand probably won't freeze at bogus inputs.

**Comment by gagoman** [comment]
👍

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**